File: user-service/src/main/java/com/example/userservice/kafka/KafkaProducer.java
package com.example.userservice.kafka;

import com.example.userservice.dto.UserLogRequest;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class KafkaProducer {
    private KafkaTemplate<String, String> kafkaTemplate;

    public KafkaProducer(KafkaTemplate<String, String> kafkaTemplate) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public UserLogRequest send(String kafkaTopic, UserLogRequest userLogRequest){
        ObjectMapper mapper = new ObjectMapper();
        String jsonInString = "";
        try {
            jsonInString = mapper.writeValueAsString(userLogRequest);
        } catch (JsonProcessingException e) {
            throw new RuntimeException(e);
        }
        kafkaTemplate.send(kafkaTopic, jsonInString);
        log.info("Kafka Producer send data from the user-service: "+ userLogRequest);
        return userLogRequest;
    }

}

File: pod-service/src/main/kotlin/miniproject.kubeBoard.podservice/kafka/KafkaProducer.kt
package miniproject.kubeBoard.podservice.kafka

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import miniproject.kubeBoard.podservice.entity.pod.req.UserLogRequest
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class KafkaProducer
    (private val kafkaTemplate: KafkaTemplate<String,String>){
    companion object {
        private val log = LoggerFactory.getLogger(KafkaProducer::class.java)
    }
    fun send(kafkaTopic: String, userLogRequest: UserLogRequest): UserLogRequest {
        val mapper = jacksonObjectMapper()
        val jsonInString: String
        try {
            jsonInString = mapper.writeValueAsString(userLogRequest)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }

        kafkaTemplate.send(kafkaTopic, jsonInString)
        log.info("Kafka Producer send data from the deployment-service: $userLogRequest")
        return userLogRequest
    }
    }
